README.md

```diff
@@ -0,0 +1,168 @@
+
+As testing was implemented late in the project, not all components have tests.
+
+To test the app in the project folder run `yarn test`.
+
+#### Tools
+
+-   **[Jest](https://jestjs.io/)** was used as the main testing platform for the app.
+-   **[Enzyme](https://airbnb.io/enzyme/docs/guides/jest.html)** library created by AirBnB to test components
+-   **Mock libraries** used:
+    -   [mock-local-storage](https://www.npmjs.com/package/jest-localstorage-mock) mock for testing localstorage
+    -   [mock-fetch](https://www.npmjs.com/package/fetch-mock) mock for testing dog list and dog image fetching
+    -   [mock-store](https://github.com/dmitry-zaets/redux-mock-store) redux store mock for async actions and middleware
+
+## Components
+
+[See a detailed list of the components used in the project.](./components.md)
+
+## Styles
+
+-   Styles were added using a vanilla stylesheet that loads inside `<App />`.
+-   [Bootstrap 4](https://getbootstrap.com/) is loaded from official CDN on index.html
+
+## Conclusion
+
+The project taught me how to use modern tools and processes such as **React**, **Redux**, **Test Driven Development** using **Jest** as a testing platform and in the next step of the project I will learn **NodeJS**.
+
+The project was started without a boilerplate, I learned how to start up a project from scratch using **Yarn**, **Webpack** and **Babel**. This project was focused on React but this knowledge can be used for any kind of project.
+
+Git is used, which I have already knew and used before, but this is the first time I use the **git-flow** workflow in a project.
+
+After configuring the project, I learned how to plan the architecture of an app, defining specs and features and how they should interact with the application and between them.
+
+As the coding started, there was so much to learn in react, starting with JSX, components and their lifecycle, props, state, the render method, render props, high order components, refs, portals, context... to name some. My mindset also changed as I have to start thinking in React. Immutability, components just serving one purpose, when and when not to create a state, how components should communicate between each other.
+
+When the app state management necessities grew, Redux was implemented with Test Driven Development process using Jest as the testing platform. This was new for me, I learned a lot from this: as to how to set up a testing environment, which helpers I would need to use and how to use them, and what to test in the code. Then some components were created using TDD with the help of the **Enzyme** library.
+
+Redux actions were used to fetch information from an API, so I learned to use **Thunk** as a middleware in these cases. I learned what actions and reducers are, and how they connect to the app, passing the store and actions as props to the components that needed them.
```

Comment:
I'd like to think you learned how to use middleware, using Thunk as an example.

Reply:
🍤